otter/convergence/planning.py

```diff
@@ -367,6 +373,103 @@ def drain_and_delete_a_server(server):
                 converge_later)
 
 
+def converge_launch_stack(desired_state, stacks):
+    """
+    Create steps that indicate how to transition from the state provided
+    by the given parameters to the :obj:`DesiredStackGroupState` described by
+    ``desired_state``.
+
+    :param DesiredStackGroupState desired_state: The desired group state.
+    :param set stacks: a set of :obj:`HeatStack` instances.
+        This must only contain stacks that are being managed for the specified
+        group.
+    :rtype: :obj:`pbag` of `IStep`
+
+    """
+    config = desired_state.stack_config
+
+    group_states = [StackState.CREATE_UPDATE_COMPLETE,
+                    StackState.CREATE_UPDATE_FAILED,
+                    StackState.CHECK_COMPLETE,
+                    StackState.CHECK_FAILED,
+                    StackState.IN_PROGRESS,
+                    StackState.DELETE_IN_PROGRESS,
+                    StackState.DELETE_FAILED]
+
+    (stacks_complete,
+     stacks_failed,
+     stacks_check_complete,
+     stacks_check_failed,
+     stacks_in_progress,
+     stacks_delete_in_progress,
+     stacks_delete_failed) = partition_groups(lambda stack: stack.get_state(),
+                                              stacks,
+                                              group_states)
+
+    stacks_amiss = (stacks_failed +
+                    stacks_check_failed +
+                    stacks_in_progress +
+                    stacks_delete_in_progress)
+
+    if stacks_delete_failed:
+        return pbag([FailConvergence(["Stacks in DELETE_FAILED found."])])
+
+    # If there are no stacks in CHECK_* or other work to be done, we assume
```

Comment:
I think it'd be nice to have a more thorough/holistic description of the workflow we put stacks through somewhere, maybe at the top of this function, or the top of this module with a `Note` (like those in service.py). The comments you have here are good but I'd like something with a bit more context. Basically, something that distills the conversation we had way back when into some documentation. ASCII art diagrams may be useful ;)